be/tool/beck.c

```diff
@@ -1836,35 +1927,30 @@ static struct cache_slot *ctg_getslot_insertcred(struct ctg_action *ca,
 	struct cache_slot *slot;
 
 	slot = cache_lookup(&b->b_cache, cas_ctg_fid);
-	if (slot == NULL) {
-		slot = cache_insert(&b->b_cache, cas_ctg_fid);
-		m0_ctg_create_credit(accum);
-		ca->cta_cid.ci_fid = ca->cta_fid;
-		m0_fid_tchange(&ca->cta_cid.ci_fid, 'T');
-		ca->cta_cid.ci_layout.dl_type = DIX_LTYPE_DESCR;
-		m0_dix_ldesc_init(&ca->cta_cid.ci_layout.u.dl_desc,
-				  &(struct m0_ext) { .e_start = 0,
-				  .e_end = IMASK_INF },
-				  1, HASH_FNC_CITY, &b->b_pver_fid);
-		m0_ctg_ctidx_insert_credits(&ca->cta_cid, accum);
-	}
+	slot = cache_insert(&b->b_cache, cas_ctg_fid);
```

Comment:
[* defect *] slot is being reinitialised before it's used. What is the use case?

Reply:
Already fixed, Please review with all commits